components/salesmate/app/salesmate.app.ts

```diff
@@ -1,13 +1,457 @@
+      label: "Last Name",
+      description: "Last name of the contact.",
+      type: "string",
+    },
+    linkedInHandle: {
+      label: "LinkedIn Handle",
+      description: "Contact's linkedin profile link.",
+      type: "string",
+    },
+    mobile: {
+      label: "Mobile",
+      description: "Mobile number of the contact.",
+      type: "string",
+    },
+    otherPhone: {
+      label: "Other Phone",
+      description: "Phone number 2 of the contact.",
+      type: "string",
+    },
+    owner: {
+      label: "Owner",
+      description: "Owner of the contact.",
+      type: "integer",
+      async options({ page }) {
+        const rows = 250;
+        const { Data } = await this.listUsers({
+          params: {
+            rows,
+            from: rows * page,
+          },
+        });
+
+        return Data.map(({
+          id: value, name: label,
+        }) => ({
+          label,
+          value,
+        }));
+      },
+    },
+    phone: {
+      label: "Phone",
+      description: "Phone number 1 of the contact.",
+      type: "string",
+    },
+    pipelineId: {
+      label: "Pipeline",
+      description: "User can select any one pipeline from pre-defined pipelines.",
+      type: "string",
+      async options({ page }) {
+        const rows = 250;
+        const { Data } = await this.listPipelines({
+          params: {
+            rows,
+            from: rows * page,
+          },
+        });
+
+        return Data.map(({ pipeline }) => (pipeline));
+      },
+    },
+    priority: {
+      label: "Priority",
+      description: "User can set the priority for the deal like high, low or medium.",
+      type: "string",
+      async options() {
+
+        const { fieldOptions } = await this.listVisibleFields({
+          module: "deal",
+          field: "priority",
+        });
+        return JSON.parse(fieldOptions).values;
+      },
+    },
+    skypeId: {
+      label: "Skype Id",
+      description: "Contact's skype id.",
+      type: "string",
+    },
+    source: {
+      label: "Source",
+      description: "User can add source from where contact came to know about user's company or deal's information like Ads, internet etc.",
+      type: "string",
+      async options() {
+
+        const { fieldOptions } = await this.listVisibleFields({
+          module: "deal",
+          field: "source",
+        });
+        return JSON.parse(fieldOptions).values;
+      },
+    },
+    stage: {
+      label: "Stage",
+      description: "User can select a stage for a deal. like new, contacted, qualified, proposal presented or in negotiation.",
+      type: "string",
+      async options({
+        page, pipeline,
+      }) {
+        const rows = 250;
+
+        const { Data } = await this.listStages({
+          pipeline,
+          params: {
+            rows,
+            from: rows * page,
+          },
+        });
+
+        return Object.keys(Data).map((key) => key)
+          .slice(0, -3);
```

Comment:
Can you add a comment describing what is being sliced here?

Reply:
Are `dealsForecastAmount`, `dealsTotalAmount`, and `dealsTotalCount` always returned in every page? Asking this because they are being sliced for every page that is being fetched.

Reply:
Actually, It doesn't even have a pagination
![image](https://user-images.githubusercontent.com/5638699/216599426-8ca523b7-89d2-400b-8886-1354e3ea60d4.png)